programs/drift/src/math/spot_withdraw.rs

```diff
@@ -136,13 +136,33 @@ pub fn check_withdraw_limits(
     Ok(valid_withdrawal)
 }
 
-pub fn calculate_availability_borrow_liquidity(spot_market: &SpotMarket) -> DriftResult<u128> {
+pub fn calculate_availability_borrow_liquidity(
+    spot_market: &SpotMarket,
+    token_amount: i128,
+) -> DriftResult<u128> {
     let deposit_token_amount = get_token_amount(
         spot_market.deposit_balance,
         spot_market,
         &SpotBalanceType::Deposit,
     )?;
 
+    let mut borrow_liquidity = 0_u128;
+    if token_amount > 0 {
+        let min_deposit_token = calculate_min_deposit_token(
+            spot_market.deposit_token_twap.cast()?,
+            spot_market.withdraw_guard_threshold.cast()?,
+        )?;
+
+        let withdraw_limit = deposit_token_amount.saturating_sub(min_deposit_token);
+
+        let token_amount = token_amount.unsigned_abs();
+        if withdraw_limit <= token_amount {
+            return Ok(withdraw_limit);
+        }
+
+        borrow_liquidity = token_amount;
```

Comment:
can you name this variable indicating its the users existing position used to "pad"